### PR TITLE
Implement new grouper client API and functions in services for adding and removing groupPath(s) as owner

### PR DESCRIPTION
**Pull request by gitCarrot**
# Ticket Link

[Ticket-1740](https://uhawaii.atlassian.net/browse/GROUPINGS-1740)

# List of squashed commits

- Implement new grouper command function to add path owners into grouping owners

- Implement the functions to call grouper api service in updateMembership and make end-points to request for adding and removing path owners

- Implements the tests for end-point in controller and integration test for TestGrouperApiService, (Make TODO tests in TestUpdateMemberService.java for separate test ticket)
- Fix code with codacy analysis
- Add and modify the comments and add the todo test

# Test Checklist

- [x] Exhibits Clear Code Structure:
- [x] Project Unit Tests Passed:
- [x] Project Integration Tests Passed:
- [x] Executes Expected Functionality:
- [x] Tested For Incorrect/Unexpected Inputs:

### CODE REVIEW

File: src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java [hunk 1]
```diff
@@ -114,6 +113,15 @@ public GroupingRemoveResult removeOwnership(String currentUser, String groupingP
         return removeOwner(currentUser, groupingPath, validIdentifier);
     }
 
+    public GroupingRemoveResults removeGroupPathOwnerships(String currentUser, String groupingPath,
+            List<String> groupPathOwner) {
+        log.info(String.format("removeOwnerships; currentUser: %s; groupingPath: %s; uhIdentifiers: %s;",
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java:118]: I believe this string should start with `removeGroupPathOwnerships`, not `removeOwnerships`.

File: src/main/java/edu/hawaii/its/api/wrapper/AddMembersCommand.java [hunk 2]
```diff
@@ -61,6 +62,21 @@ private AddMembersCommand addUid(String uid) {
         return this;
     }
 
+    public AddMembersCommand addGroupPathOwner(String groupPath) {
+        WsSubjectLookup wsSubjectLookup = new WsSubjectLookup();
+        wsSubjectLookup.setSubjectSourceId("g:gsa");
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/wrapper/AddMembersCommand.java:67]: Maybe a comment on line 67 would be helpful for others to understand why we need `g:gsa`.

File: src/main/java/edu/hawaii/its/api/wrapper/RemoveMembersCommand.java [hunk 3]
```diff
@@ -46,6 +47,22 @@ private RemoveMembersCommand addUid(String uid) {
         return this;
     }
 
+    public RemoveMembersCommand addGroupPathOwner(String groupPath) {
+        WsSubjectLookup wsSubjectLookup = new WsSubjectLookup();
+        wsSubjectLookup.setSubjectSourceId("g:gsa");
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/wrapper/RemoveMembersCommand.java:52]: Same comment here.

File: src/test/java/edu/hawaii/its/api/service/TestUpdateMemberService.java [hunk 4]
```diff
@@ -442,6 +442,16 @@ public void checkIfSelfOptOrAdminTest() {
...
 
+//    @Test
+//    public void addRemovePathOwnershipsTest(){
+//        // Todo integration test for updating grouping owners with path owners
+//    }
+
+//    @Test
+//    public void checkAllMembersAfterAddRemovePathOwnershipsTest(){
+//        // Todo integration test for checking the changes of members' group after adding and removing path owners
+//    }
```
Comment (JorWo) [inline src/test/java/edu/hawaii/its/api/service/TestUpdateMemberService.java:453]: If you are planning to add the integration tests in another ticket, could you at the very least add tests for the wrapper classes (`AddMembersCommand.java` amd `RemoveMembersCommand.java`) and an integration test inside `TestGroupingsRestControllerv2_1.java`? Thanks.

Comment (gitCarrot) [reply src/test/java/edu/hawaii/its/api/service/TestUpdateMemberService.java:453]: I just added just TODO tests inside of the TestGroupingsRestControllerv2_1.java.

**Review by JorWo — APPROVED** [review]

**Pull request merged**